### ISSUE TITLE
Merge Queues

### ISSUE THREAD

**Issue by wrtobin** [GEOS-DEV/GEOS#2462]
Do we want to implement an actual merge train / merge queue solution to prevent the situation we often have now where branches in the PR Merge Order > Merge Queue have to merge-in previous merges and rerun the entire CI suite?

We would need to be more careful about merge order, and people would need to be aware of everything ahead of them in the merge queue to prevent possibly merge conflicts, etc, so there is a slightly higher demand on developers, but if used properly could improve time-to-merge and reduce overall CI usage.

**Comment by wrtobin** [comment]
... I suppose having CI that doesn't have issues with timing out is probably a prerequisite of this.

**Comment by untereiner** [comment]
What about github's merge queue beta functionality ?

**Comment by wrtobin** [comment]
Yeah that is more or less what this is about. It will requires discussions and planning here at Livermore though, and will require developers have a slightly different understanding of how the merge queue operates.

**Comment by untereiner** [comment]
@TotoGaz @rrsettgast github merge queue is [generally available ](https://github.blog/2023-07-12-github-merge-queue-is-generally-available/). Would it be possible for you to  enable this new branch protection ? And, only this one to put all accepted PR in the queue automatically ?
It would also help in the process of continuous delivery #3035
